src/components/utils/blocks.ts

```diff
@@ -119,3 +137,98 @@ export function convertStringToBlockData(stringToImport: string, conversionConfi
     return {};
   }
 }
+
+/**
+ * Returns list of all available conversion menu items
+ *
+ * @param currentBlock - current block we need to get convetion data for
+ * @param editorModules - access to editor modules
+ */
+export async function getConvertToItems(currentBlock: Block, editorModules: EditorModules): Promise<PopoverItemDefaultParams[]> {
+  const conversionEntries = Array.from(editorModules.Tools.blockTools.entries());
+
+  const resultItems: PopoverItemDefaultParams[] = [];
+
+  const blockData = await currentBlock.data;
+
+  conversionEntries.forEach(([toolName, tool]) => {
+    const conversionConfig = tool.conversionConfig;
+
+    /**
+     * Skip tools without «import» rule specified
+     */
+    if (!conversionConfig || !conversionConfig.import) {
+      return;
+    }
+
+    tool.toolbox?.forEach((toolboxItem) => {
+      /**
+       * Skip tools that don't pass 'toolbox' property
+       */
+      if (isEmpty(toolboxItem) || !toolboxItem.icon) {
+        return;
+      }
+
+      let shouldSkip = false;
+
+      if (toolboxItem.data !== undefined) {
+        /**
+         * When a tool has several toolbox entries, we need to make sure we do not add
+         * toolbox item with the same data to the resulting array. This helps exclude duplicates
+         */
+        const hasSameData = isSameBlockData(toolboxItem.data, blockData);
+
+        shouldSkip = hasSameData;
+      } else {
+        shouldSkip = toolName === currentBlock.name;
+      }
+
+
+      if (shouldSkip) {
+        return;
+      }
+
+      resultItems.push({
+        icon: toolboxItem.icon,
+        title: toolboxItem.title,
+        name: toolName,
+        closeOnActivate: true,
+        onActivate: () => {
+          const { BlockManager, BlockSelection, Caret } = editorModules;
+
+          BlockManager.convert(currentBlock, toolName, toolboxItem.data);
+
+          BlockSelection.clearSelection();
+
+          window.requestAnimationFrame(() => {
+            Caret.setToBlock(currentBlock, Caret.positions.END);
+          });
```

Comment:
also unexpected logic for util

Reply:
can't figure out how and where to move this logic